Add pcb release command

akhilles opened the pull request:
Intentially spartan in terms of flags/arguments. All that's needed is the path to the .zen file.

To test:
```
akhilles@evil ~/s/d/pcb (pcb/release)> cargo run -- release crates/pcb/test-workspace/boards/TestBoard.zen
   Compiling pcb v0.2.0-prerelease.2 (/Users/akhilles/src/diode/pcb/crates/pcb)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.97s
     Running `target/debug/pcb release crates/pcb/test-workspace/boards/TestBoard.zen`
✓ Release information gathered

Release Metadata
{
  "git": {
    "describe": "85c1a66",
    "workspace": "/Users/akhilles/src/diode/pcb/crates/pcb/test-workspace"
  },
  "release": {
    "created_at": "2025-08-14T22:37:53.167657+00:00",
    "layout_path": "/Users/akhilles/src/diode/pcb/crates/pcb/test-workspace/boards/build/TestBoard",
    "staging_directory": "/Users/akhilles/src/diode/pcb/crates/pcb/test-workspace/boards/.pcb/releases/TestBoard/85c1a66",
    "version": "85c1a66",
    "workspace_root": "/Users/akhilles/src/diode/pcb/crates/pcb/test-workspace",
    "zen_file": "/Users/akhilles/src/diode/pcb/crates/pcb/test-workspace/boards/TestBoard.zen"
  },
  "system": {
    "arch": "aarch64",
    "cli_version": "0.2.0-prerelease.2",
    "platform": "macos",
    "user": "akhilles"
  }
}
✓ Copying source files and dependencies
✓ Copying layout files
✓ Generating unmatched BOM
✓ Writing release metadata
✓ Creating release archive

✓ Release 85c1a66 staged successfully
Archive: /Users/akhilles/src/diode/pcb/crates/pcb/test-workspace/boards/.pcb/releases/TestBoard/85c1a66.zip
```

Implemented:
- Workspace root detection (even without pcb.toml)
- Git-based versioning (e.g. tag > commit hash > dirty commit hash)
- Zen source files copied into release dir (release-dir/src)
- Dependencies are vendored into release dir (release-dir/src/vendor)
- Unmatched bom is generated into release dir (release-dir/bom)
- KiCad layout dir is copied into release dir (release-dir/layout)
- Metadata file is generated into release dir (release-dir/metadata.json)
- Generated artifacts are stored in .pcb/ dir (for now) and not uploaded

TODO:
- Upload the release bundle to API server
- Add a --dry-run to generate release bundle but not upload (similar to cargo)
- Use the vendor/ dir for dependency resolution for deterministic builds
- Add pcb vendor subcommand, --offline flag for build/layout
- Add more release artifacts (e.g. gerbers, odb, etc.)

LK reviewed (commented):
This is so sick!!

I get an error when I try to release a file that uses workspace-relative (`//`) loads:

```
Error: Zen file evaluation failed:
Error: /Users/lenny/code/diode/registry/modules/robotics/HighPowerRelay.zen:20:11-53 Failed to resolve module path '//components/CP1M12V/CP1M12V.zen': File not found: /Users/lenny/code/diode/registry/modules/robotics/components/CP1M12V/CP1M12V.zen
```
(I updated the HighPowerRelay.zen file to use workspace-relative paths to test)

Also instead of putting the .pcb folder wherever the user/file is, can we put it where the workspace root is? We already symlink in the cache files there so its a natural place to dump these too.

akhilles commented:
> I get an error when I try to release a file that uses workspace-relative (`//`) loads:
> 
> ```
> Error: Zen file evaluation failed:
> Error: /Users/lenny/code/diode/registry/modules/robotics/HighPowerRelay.zen:20:11-53 Failed to resolve module path '//components/CP1M12V/CP1M12V.zen': File not found: /Users/lenny/code/diode/registry/modules/robotics/components/CP1M12V/CP1M12V.zen
> ```

Ah, didn't know this was possible. Will fix. Is it safe to assume workspace-relative loads are only possible when there's a `pcb.toml` file? Otherwise, it gets kinda chicken-and-eggy to determine what the workspace root is.

> Also instead of putting the .pcb folder wherever the user/file is, can we put it where the workspace root is? We already symlink in the cache files there so its a natural place to dump these too.

Yes, will make this change. Currently, when there's no `pcb.toml` file, the `.pcb/cache` dir is created in the parent dir of the `.zen` file (not the workspace root): `crates/pcb/test-workspace/boards/.pcb`. So, that's why I added releases to the same dir. But I think `pcb build` should be updated to have a consistent understanding of workspace root, so that it uses `crates/pcb/test-workspace/.pcb` as well.

akhilles commented:
@LK addressed all the feedback:

- workspace-relative loads now work (assuming there's a pcb.toml)
- releases are always staged in workspace root .pcb dir
